Extension/src/LanguageServer/lmTool.ts

```diff
@@ -44,6 +47,85 @@ const knownValues: { [Property in keyof ChatContextResult]?: { [id: string]: str
     }
 };
 
+function formatChatContext(chatContext: ChatContextResult): void {
+    type KnownKeys = 'language' | 'standardVersion' | 'compiler' | 'targetPlatform' | 'targetArchitecture';
+    for (const key in knownValues) {
+        const knownKey = key as KnownKeys;
+        if (knownKey && knownValues[knownKey] && chatContext[knownKey]) {
+            chatContext[knownKey] = knownValues[knownKey][chatContext[knownKey] as string] || chatContext[knownKey];
+        }
+    }
+}
+
+export interface CppContext {
+    language: string;
+    standardVersion: string;
+    compiler: string;
+    targetPlatform: string;
+    targetArchitecture: string;
+    compilerArguments: string;
+}
+
+// To be updated after A/B experiments.
+const defaultCompilerArgumentFilters: { [id: string]: RegExp | undefined } = {

```

Comment:
nit: `undefined` means the key doesn't actually exist in the object.  This might be defined like this instead:
```ts
const defaultCompilerArgumentFilters: { [id: string]: RegExp } = {};
```

Reply:
(though from what I can tell, this object never adds defaults from any of the compilers and will remain empty)

Reply:
Updated. Will update defaults after A/B experiemnts, we'll know better on what arguments we should provide to help completions.